adapters/pubmatic/pubmatic.go

```diff
@@ -368,6 +368,73 @@ func (a *PubmaticAdapter) MakeRequests(request *openrtb.BidRequest) ([]*adapters
 	}}, errs
 }
 
+// validateAdslot validate the optional adslot string
+// valid formats are 'adslot@WxH', 'adslot' and no adslot
+func validateAdSlot(adslot string, imp *openrtb.Imp) error {
+	adSlotStr := strings.TrimSpace(adslot)
+
+	if len(adSlotStr) == 0 {
+		return nil
+	}
+
+	if !strings.Contains(adSlotStr, "@") {
+		imp.TagID = adSlotStr
+		return nil
+	}
+
+	adSlot := strings.Split(adSlotStr, "@")
+	if len(adSlot) == 2 && adSlot[0] != "" && adSlot[1] != "" {
+		imp.TagID = strings.TrimSpace(adSlot[0])
```

Comment:
Do we need to call `TrimSpace` after every `strings.Split` call? There shouldn't be any spaces since you already do a `TrimSpace` on the original `adSlot` string itself, right?

Reply:
In most cases, the adslot string will be like "test@WxH" and in those cases, TrimSpace is not required more than once but in the rare case where adslot string, for some reason, is something like " test@ 34 x 45 ", TrimSpace called on the entire string will only handle the leading and trailing spaces but when the string is split, the width will be "34 " and height "45 ". For those cases when I have retained the TrimSpace after the spilts.

But there are two places where we can do without TrimSpace and I will remove it there as we handle spaces for individual elements (width and height)